server/src/main/java/io/crate/planner/operators/Eval.java

```diff
@@ -126,7 +126,15 @@ public FetchRewrite rewriteToFetch(TableStats tableStats, Collection<Symbol> use
         }
         for (Symbol output : outputs) {
             newReplacedOutputs.put(output, mapToFetchStubs.apply(output));
-            SymbolVisitors.intersection(output, newSource.outputs(), newOutputs::add);
+
+            // If output is an alias, unwrap it so that we can compare "real" symbol with newSource.outputs
+            var unwrappedOutput = output.accept(AliasResolver.INSTANCE, null);
```

Comment:
or can use `SymbolVisitors.any(output::equals, newSource.outputs()))` like we already [did](https://github.com/crate/crate/commit/f05699608adeb60b98577ce309b049151b82618b#diff-7b421ef17c8959821108907844fff299dd8e93e2f3a8212d7e4e15c85424f6d0R381) in `Collect` when faced an issue with unwrapping aliases. This way we don't have to pass a `Consumer` and can control here what we want to add into `newOutputs`.

Reply:
I am not sure if this scenario is possible, but if  `outputs` contains scalars which are not contained in `newSource.outputs()`, then this code below would probably break and the `SymbolVisitors.any(...)` approach would probably cover it, so might be safer to go with that?

Reply:
used any(), replaced itest with a planner test and added a change entry in https://github.com/crate/crate/pull/13439/commits/310dce6b2d314fd5f86acfb7dec8673b44b9bf83